URL link checks may break due to unhandled exceptions

Example run: https://github.com/SaschaMann/v3/runs/1447193663#step:6:30

```
Unhandled exception. System.AggregateException: One or more errors occurred. (An error occurred while sending the request.)
 ---> System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: The response ended prematurely.
   at System.Net.Http.HttpConnection.FillAsync(Boolean async)
   at System.Net.Http.HttpConnection.ReadNextResponseHeaderLineAsync(Boolean async, Boolean foldedHeadersAllowed)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 337
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 870
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 878
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 1142
   at MarkdownLinkChecker.Checker.linkStatusForDocuments(Options options, Document[] documents) in /home/runner/work/MarkdownLinkChecker/MarkdownLinkChecker/src/MarkdownLinkChecker/Checker.fs:line 73
   at MarkdownLinkChecker.Checker.checkDocuments(Options options, Document[] documents) in /home/runner/work/MarkdownLinkChecker/MarkdownLinkChecker/src/MarkdownLinkChecker/Checker.fs:line 129
   at MarkdownLinkChecker.Program.main(String[] argv) in /home/runner/work/MarkdownLinkChecker/MarkdownLinkChecker/src/MarkdownLinkChecker/Program.fs:line 16
/home/runner/work/_temp/fa4c1373-c40e-4010-9d9b-ff593039375a.sh: line 14:  2727 Aborted                 (core dumped) markdownlinkchecker -v detailed -f $files
Error: Process completed with exit code 134.
```

I don't know what link in particular caused that.

--- a/src/MarkdownLinkChecker/Checker.fs
+++ b/src/MarkdownLinkChecker/Checker.fs
@@ -10,6 +10,7 @@ open MarkdownLinkChecker.Options
 type LinkStatus =
     | Found
     | NotFound
+    | Error
 
 type Status =
     | Valid
@@ -58,12 +59,14 @@ let private checkFileStatus (path: string) =
 
 let private checkLinkStatus (link: Link) =
     async {
-        let! status =
+        let! result =
             match link with
-            | UrlLink (_) -> checkUrlStatus (linkKey link)
-            | FileLink (_) -> checkFileStatus (linkKey link)
-
-        return (link, status)
+            | UrlLink (_) -> checkUrlStatus (linkKey link) |> Async.Catch
+            | FileLink (_) -> checkFileStatus (linkKey link) |> Async.Catch
+   
+        match result with
+        | Choice1Of2 status -> return (link, status)
+        | Choice2Of2 _ -> return (link, Error)
     }
 
 let private checkLinkStatuses (documents: Document []) =
@@ -96,7 +99,7 @@ let private checkedDocumentStatus (checkedLinks: CheckedLink []) =
 let private logCheckedDocument (options: Options) (checkedDocument: CheckedDocument) =
     let invalidLinksCount =
         checkedDocument.CheckedLinks
-        |> Seq.filter (fun checkedLink -> checkedLink.Status = NotFound)
+        |> Seq.filter (fun checkedLink -> checkedLink.Status <> Found)
         |> Seq.length
 
     let filePath =
@@ -115,9 +118,13 @@ let private logCheckedDocument (options: Options) (checkedDocument: CheckedDocum
         let position = linkPosition checkedLink.Link
         let reference = linkReference checkedLink.Link
 
-        if checkedLink.Status = Found
-        then options.Logger.Detailed(sprintf "✅ (%d,%d): %s" position.Line position.Column reference)
-        else options.Logger.Minimal(sprintf "❌ (%d,%d): %s" position.Line position.Column reference)
+        match checkedLink.Status with
+        | Found ->
+            options.Logger.Detailed(sprintf "✅ (%d,%d): %s" position.Line position.Column reference)
+        | NotFound ->
+            options.Logger.Minimal(sprintf "❌ (%d,%d): %s" position.Line position.Column reference)
+        | Error ->
+            options.Logger.Minimal(sprintf "❌ (%d,%d): %s (error)" position.Line position.Column reference)
 
 let private checkDocument (options: Options) (checkedLinks: Map<string, LinkStatus>) (document: Document) =
     let checkedLinks = toCheckedLinks checkedLinks document

